core/sql/executor/ExHdfsScan.cpp

```diff
@@ -585,6 +585,17 @@ ExWorkProcRetcode ExHdfsScanTcb::work()
                    openType //
                    );
                 
+                if ((retcode < 0) &&
+                    ((errno == ENOENT) || (errno == EAGAIN)))
```

Comment:
errno is a global variable that might be set by any system library call. It is dangerous to rely on it except right after the system call where it is set. But in this code, it looks like the library call is buried inside ExpLOBInterfaceSelectCursor. It would be safer if the latter function saved errno after whatever system library call it calls, and returns that as an "out" parameter, which this code could then check.